src/Swapper1InchV6.sol

```diff
@@ -0,0 +1,121 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.25;
+
+/**
+ * @notice 1Inch Pathfinder V6 implementation of the general ISwapper interface.
+ * @author ether.fi [shivam@ether.fi]
+ * @dev It is possible that dust token amounts are left in this contract after a swap.
+ * This can happen with some tokens that don't send the full transfer amount.
+ * These dust amounts can build up over time and be used by anyone who calls the `swap` function.
+ */
+import {SafeERC20, IERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IAggregationExecutor, IOneInchRouterV6, SwapDescription} from "./interfaces/IOneInch.sol";
+import {ISwapper} from "./interfaces/ISwapper.sol";
+
+contract Swapper1InchV6 is ISwapper {
```

Comment:
Is it the implementation from scratch or is there any similar reference we can send to reduce the audit complexity?

Reply:
This I implemented from Origin dollar repo. They had it for v5, I implemented for v6

https://github.com/OriginProtocol/origin-dollar/blob/master/contracts/contracts/swapper/Swapper1InchV5.sol

Reply:
Nice. is this already audited? if yes, then let's say that in the code comment

Reply:
Yes its already audited. Sure will add a comment.